src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/PropertySetAnalysis/PropertySetAnalysis.PropertySetAbstractValueDomain.cs

```diff
@@ -23,18 +24,68 @@ private class PropertySetAbstractValueDomain : AbstractValueDomain<PropertySetAb
 
             public override int Compare(PropertySetAbstractValue oldValue, PropertySetAbstractValue newValue, bool assertMonotonicity)
             {
-                return Comparer<PropertySetAbstractValue>.Default.Compare(oldValue, newValue);
+                Debug.Assert(oldValue != null);
+                Debug.Assert(newValue != null);
+
+                if (Object.ReferenceEquals(oldValue, newValue))
+                {
+                    return 0;
+                }
+
+                int maxKnownCount = Math.Max(oldValue.KnownValuesCount, newValue.KnownValuesCount);
+                int result = 0;
+                for (int i = 0; i < maxKnownCount; i++)
+                {
+                    if (oldValue[i] == newValue[i])
+                    {
+                        continue;
+                    }
+                    else if (oldValue[i] < newValue[i])
+                    {
+                        result = -1;
+                    }
+                    else
+                    {
+                        Debug.Assert(oldValue[i] > newValue[i]);
```

Comment:
Nit: Do we need this assert? Seems evident from previous two conditional checks.

Reply:
Nah, sometimes I like to leave asserts to explicitly label the case of an else block, but I guess this case seems pretty trivial.